namespace/namespace.go

```diff
@@ -46,17 +40,7 @@ func MustNew(version uint8, id []byte) Namespace {
 
 // NewFromBytes returns a new namespace from the provided byte slice.
 func NewFromBytes(bytes []byte) (Namespace, error) {
-	if len(bytes) != NamespaceSize {
-		return Namespace{}, fmt.Errorf("invalid namespace length: %v must be %v", len(bytes), NamespaceSize)
-	}
-
-	err := validateVersionSupported(bytes[VersionIndex])
-	if err != nil {
-		return Namespace{}, err
-	}
-
-	err = validateID(bytes[VersionIndex], bytes[NamespaceVersionSize:])
-	if err != nil {
+	if err := Validate(bytes[VersionIndex], bytes[NamespaceVersionSize:]); err != nil {
```

Comment:
nit: to avoid confusing byte slicing you could do:
```go
ns := Namespace{data: bytes}
err := Validate(ns.Version(), ns.ID())
if err != nil {
   return Namespace{}, err
}

return ns
```

Reply:
Still, I'm not sure why we wouldn't add Validate as a method. (I know it's guaranteed to be valid after the construction but still)

Reply:
It gets used by the `share` package

Reply:
:merge-it: